src/plain_printer.cc

```diff
@@ -72,41 +72,62 @@ std::vector<std::string> champsim::plain_printer::format(CACHE::stats_type stats
   using mshr_merge_value_type = typename decltype(stats.mshr_merge)::value_type;
   using mshr_return_value_type = typename decltype(stats.mshr_return)::value_type;
 
+  std::vector<std::size_t> cpus;
+
+  //build a vector of all existing cpus
+  auto stat_keys = {stats.hits.get_keys(), stats.misses.get_keys(), stats.mshr_merge.get_keys(), stats.mshr_return.get_keys()};
+  for (auto keys : stat_keys)
+    for (auto key : keys)
+      if(std::find_if(cpus.begin(), cpus.end(), [key=key](const auto val) {return key.second == val;}) == cpus.end())
+        cpus.push_back(key.second);
+        
+  //ensure stats were collected for active cpu, print warning otherwise
```

Comment:
You note here that it ensures stats were printed for the "active" cpu, but this is not quite the case. It is the most recently used cpu whenever the phase began. It seems like it's a bit of a hack that works for caches that are connected only to one core, but for caches downstream of more than one core It doesn't actually accomplish what you want.